From 01210abf512bc1ced1ca8df8495df20b5fd98658 Mon Sep 17 00:00:00 2001
From: Kiran Gunda <kgunda@codeaurora.org>
Date: Thu, 16 Apr 2015 18:39:14 +0530
Subject: [PATCH 758/959] IKSWM-24183: spi_qsd: Use polling option while
 disconnecting the BAM pipe

Change-Id: I6f591538ac660fcc8ec90ac82ba6ee4666d4770f
Signed-off-by: Kiran Gunda <kgunda@codeaurora.org>
Reviewed-on: http://gerrit.mot.com/736007
Submit-Approved: Jira Key <jirakey@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Reviewed-by: Connie Zhao <czhao1@motorola.com>
SLTApproved: Connie Zhao <czhao1@motorola.com>
Reviewed-on: http://gerrit.mot.com/830302
Reviewed-by: Igor Kovalenko <igork@motorola.com>
---
 drivers/spi/spi_qsd.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/spi/spi_qsd.c b/drivers/spi/spi_qsd.c
index c25c06c8206..d902707e7ad 100644
--- a/drivers/spi/spi_qsd.c
+++ b/drivers/spi/spi_qsd.c
@@ -801,7 +801,15 @@ static void msm_spi_set_mx_counts(struct msm_spi *dd, u32 n_words)
 static int msm_spi_bam_pipe_disconnect(struct msm_spi *dd,
 						struct msm_spi_bam_pipe  *pipe)
 {
-	int ret = sps_disconnect(pipe->handle);
+	int ret = 0;
+	struct sps_connect config = pipe->config;
+	config.options |= SPS_O_POLL;
+	ret = sps_set_config(pipe->handle, &config);
+	if (ret) {
+		pr_err("sps_set_config() failed ret %d\n", ret);
+		return ret;
+	}
+	ret = sps_disconnect(pipe->handle);
 	if (ret) {
 		dev_dbg(dd->dev, "%s disconnect bam %s pipe failed\n",
 							__func__, pipe->name);
-- 
2.11.0

